### DISCUSSION TITLE
export usePagination

### DISCUSSION THREAD

**Discussion by thany** [vuetifyjs/vuetify#20035] [General]
Seems like a useful hook.

I have a usecase where I need to build a totally custom pagination component, and it seems like it'd be very useful to hook into the existing pagination of the VDataTable I'm making this pagination for. But this hook doesn't seem to be exported.

1. What is the reasoning behind not exporting useful hooks, such as this one?
2. Was this a deliberate decision in the first place, or was it just forgotten to be exported?
3. What is the recommended way to create custom pagination, without having access to this super useful hook?

For reference, it's used like this:
https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/components/VDataTable/VDataTableFooter.tsx

And I need to create something like it, but not exactly it. The customisations for the VDataTableFooter do not cover the usecase, and adding props to cover my (and every other) usecase would be far more difficult for the Vuetify devs.

**Comment by kor3k** [comment]
it is somewhat partially exposed to slot props

https://vuetifyjs.com/en/api/v-data-table/#slots-bottom

`page, itemsPerPage, pageCount, setItemsPerPage` are, but `itemsLength` is not :-/
so not everything is exposed.

it would be nice if the whole yield of `usePagination()` is exposed as a method/prop on the table component instance.
